source/brailleDisplayDrivers/brailliantB.py

```diff
@@ -190,11 +190,25 @@ def __init__(self, port="auto"):
 
 	def _initAttempt(self):
 		if self.isHid:
+			# Ensure to set self.numCells only at the end of this method to prevent display writes before the capabilities/numCells request
+			numCells = 0
+			# First try to get cell count from _writeSize
+			try:
+				# _writeSize includes 4 bytes of overhead, so subtract 4
+				numCells = self._dev._writeSize - 4
+			except AttributeError:
+				log.debugWarning("Could not get _writeSize from HID device")
+
+			# Adjust numCells based on reported number of cells
 			try:
 				data: bytes = self._dev.getFeature(HR_CAPS)
+				reportedNumCells = data[24]
+				if reportedNumCells > 0:
+					# Update numCells based on reported cell count from the device
+					numCells = reportedNumCells
```

Comment:
```suggestion
					numCells = reportedNumCells
				else:
					log.debugWarning("Could not get number of cells from HID device using HR_CAPS")
```

Reply:
Logging that we did not get a numCells from HR_CAPS is fine with me, but I'm not sure if it should be a debugWarning if this is or will be expected behavior. Since I don't have official protocol documentation, I'm not sure about that.

Reply:
After some more thought, based on other things we log at the debugWarning level, I think this is appropriate.